engine/src/mode_development.cpp

```diff
@@ -554,7 +559,34 @@ bool MCObjectListAppend(MCObjectList *&x_list, MCObject *p_object, bool p_unique
 	return true;
 }
 
-bool MCObjectListAppend(MCObjectList *&x_list, MCObjectList *p_list, bool p_unique)
+bool MCObjectListAppendObjectAndBehaviors(MCObjectList *&x_list, MCObject *p_object)
+{
+    if (!MCObjectListAppendObject(x_list, p_object))
+    {
+        return false;
+    }
+    
+    MCParentScript *t_parent_script = p_object->getparentscript();
+    while(t_parent_script != nullptr)
+    {
+        if (t_parent_script->IsBlocked())
+        {
+            continue;
```

Comment:
Doh! Good point. Probably easiest to just invert the logic - only append the object if not blocked.